### ISSUE TITLE
test_server_partition_write  failed on master

### ISSUE THREAD

**Issue by sunxiaoguang** [tikv/tikv#2863]
raftstore::test_transport::test_server_partition_write

CI failed a test not related to pull request.  

CI task is here: https://circleci.com/gh/pingcap/tikv/8917#config/containers/0

test raftstore_cases::test_transport::test_server_partition_write ... thread 'raftstore_cases::test_transport::test_server_partition_write' panicked at 'called `Result::unwrap()` on an `Err` value: Other(StringError("[tests/raftstore/cluster.rs:261]: can\'t get leader of region 1"))', libcore/result.rs:916:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:68
             at libstd/sys_common/backtrace.rs:57
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:380
   3: std::panicking::default_hook
             at libstd/panicking.rs:396
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:576
   5: std::panicking::begin_panic
             at libstd/panicking.rs:537
   6: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:521
   7: rust_begin_unwind
             at libstd/panicking.rs:497
   8: core::panicking::panic_fmt
             at libcore/panicking.rs:71
   9: core::result::unwrap_failed
             at /checkout/src/libcore/macros.rs:23
  10: <core::result::Result<T, E>>::unwrap
             at /checkout/src/libcore/result.rs:782
  11: <integrations::raftstore::cluster::Cluster<T>>::request
             at tests/raftstore/cluster.rs:550
  12: <integrations::raftstore::cluster::Cluster<T>>::get_impl
             at tests/raftstore/cluster.rs:603
  13: <integrations::raftstore::cluster::Cluster<T>>::must_get
             at tests/raftstore/cluster.rs:599
  14: integrations::raftstore_cases::test_transport::test_partition_write
             at tests/raftstore_cases/test_transport.rs:41
  15: integrations::raftstore_cases::test_transport::test_server_partition_write
             at tests/raftstore_cases/test_transport.rs:63
  16: <F as alloc::boxed::FnBox<A>>::call_box
             at libtest/lib.rs:1440
             at /checkout/src/libcore/ops/function.rs:223
             at /checkout/src/liballoc/boxed.rs:817
  17: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
FAILED

**Comment by breeswish** [comment]
Some tests are not very stable as far as I know. Usually we invoke a rebuild.. Let's keep this issue open so that it can be further improved.

**Comment by sunxiaoguang** [comment]
Ok, I see. thanks.

**Comment by BusyJay** [comment]
The failure occurred 10 days ago was fixed by #8971. The failure occurred 5 days ago seemed to be a bug of system that assigned same port to two listening requests.
```
2020/11/11 05:08:03.630 server.rs:226: [INFO] listening on addr, addr: 127.0.0.1:10146
2020/11/11 05:08:03.717 cluster.rs:284: [DEBG] node 5 started
--
2020/11/11 05:08:04.093 server.rs:226: [INFO] listening on addr, addr: 127.0.0.1:10146
2020/11/11 05:08:04.202 cluster.rs:284: [DEBG] node 1 started
```
10146 was assigned twice to different stores.

I'm not sure how this happens, but I'm going to use a global port allocator to avoid the conflict.
